### PR TITLE
Add testing for security service of edgeX.

### PR DESCRIPTION
**Pull request by tingyuz**
Signed-off-by: Tingyu Zeng <tingyu.zeng@rsa.com>

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -58,6 +61,104 @@ services:
     - volume
     - consul
 
+  vault:
+    image: edgexfoundry/docker-edgex-vault:0.2.0
```
Comment (JPWKU) [inline docker-compose.yml]: Need to parameterize this and pull the image from nexus3.edgexfoundry.org:10004, which is the staging repo.
The parameters are conditionally set by `bin/env.sh` and `bin/arm64_env.sh`.  Note the docker image names noting arm64 available in nexus.
This way we can be sure that we are always testing the latest image available for both x86 and arm architecture.

File: docker-compose.yml [hunk 2]
```diff
@@ -58,6 +61,104 @@ services:
...
+    volumes:
+      - vault-config:/vault/config
+      - vault-file:/vault/file
+      - vault-logs:/vault/logs
+    depends_on:
+      - volume
+      - consul
+
+  vault-worker:
+    image: edgexfoundry/docker-edgex-vault-worker:latest
```
Comment (JPWKU) [inline docker-compose.yml]: Parameterize same as above comment.

File: docker-compose.yml [hunk 3]
```diff
@@ -58,6 +61,104 @@ services:
...
+        - 'KONG_PROXY_ACCESS_LOG=/dev/stdout'
+        - 'KONG_ADMIN_ACCESS_LOG=/dev/stdout'
+        - 'KONG_PROXY_ERROR_LOG=/dev/stderr'
+        - 'KONG_ADMIN_ERROR_LOG=/dev/stderr'
+        - 'KONG_ADMIN_LISTEN=0.0.0.0:8001, 0.0.0.0:8444 ssl'
+    depends_on:
+        - kong-db
+
+  edgex-proxy:
+    image: "edgexfoundry/docker-edgex-proxy-go:latest"
```
Comment (JPWKU) [inline docker-compose.yml]: Parameterize same as above comment.

**Review by JPWKU — CHANGES_REQUESTED** [review]
Need to address this small issue first.

File: bin/arm64_env.sh [hunk 4]
```diff
@@ -17,4 +17,11 @@ export supportRulesengine=nexus3.edgexfoundry.org:10004/docker-support-rulesengi
 export exportClient=nexus3.edgexfoundry.org:10004/docker-export-client-go-arm64:1.0.0
 export exportDistro=nexus3.edgexfoundry.org:10004/docker-export-distro-go-arm64:1.0.0
 export deviceVirtual=nexus3.edgexfoundry.org:10004/docker-device-virtual-arm64:0.6.0
+
+export vault=nexus3.edgexfoundry.org:10004/docker-edgex-vault:0.2.0
+export vaultWorker=nexus3.edgexfoundry.org:10004/docker-edgex-vault-worker:latest
+export kongdb=postgres:9.5
+export kong=kong:0.13.0
+export edgexProxy=nexus3.edgexfoundry.org:10004/docker-edgex-proxy-go:latest
+
```
Comment (JPWKU) [inline bin/arm64_env.sh]: For Lines 21 through 25 in this file we need to point at the arm64 versions of these services.  The kong and postgres images may be multi architecture, I'm not sure.  But the EdgeX images will have `-arm64` in the image name.

Comment (JPWKU) [reply bin/arm64_env.sh]: `nexus3.edgexfoundry.org:10004/docker-edgex-vault-worker-go-arm64:0.3.0`
`nexus3.edgexfoundry.org:10004/docker-edgex-vault-arm64:0.3.0`
`nexus3.edgexfoundry.org:10004/docker-edgex-proxy-go-arm64:0.2.1`

I got those image names and tags from here https://nexus3.edgexfoundry.org/#browse/browse:docker.staging
If you see an image tag that should be there but is not reconciled with the security repo master tag please let me know.

**Review by JPWKU — CHANGES_REQUESTED** [review]

File: bin/arm64_env.sh (hunk 4 above)
Comment (tingyuz) [reply bin/arm64_env.sh]: Updated the file for arm.

Comment (JPWKU) [reply bin/arm64_env.sh]: I'm not seeing those changes in the file @tingyuz , I may be missing it though.

Comment (tingyuz) [reply bin/arm64_env.sh]: my fault... just recheck in the file and trigger the build.

**Review by JPWKU — APPROVED** [review]
LGTM pending verify

**Comment by cloudxxx8** [comment]
@tingyuz why is this PR closed?

**Comment by tingyuz** [comment]
> @tingyuz why is this PR closed?

I pushed master to my feature branch by accident so Git closed the PR automatically as they are exactly same. Please use PR #160 instead.

**Pull request closed without merging**